src/yarpWholeBodySensors.cpp

```diff
@@ -938,28 +1267,150 @@ bool yarpWholeBodySensors::convertIMU(double * wbi_imu_readings, const double *
+            // Process IMU accelerometer data
+            
+            yarp::sig::Vector *vt;// = portsMTBSensors[accelerometer_runtime_index]->read(wait);
+            vt = portsMTBSensors[accelerometer_runtime_index]->read(wait);
+    
+//             std::cout<<"----------------Made a read on the port\n";
+//             std::cout<<"MTBLastRead size : "<<mtbLastRead.size()<<", accelerometer_runtime_index = "<<accelerometer_runtime_index<<"\n";
+//             std::cout<<"VT size: "<<vt->size()<<"\n";
+            if(vt!=NULL) {
+                if(mtbLastRead[accelerometer_runtime_index].size()==0)
+                {
+//                     std::cout<<" Trying to reset MTBLastRead size \n";
+                    mtbLastRead[accelerometer_runtime_index].resize(vt->size(),0.0);
+                }
+//                 std::cout<<"MTBLastRead size : "<<mtbLastRead[accelerometer_runtime_index].size()<<", data vt size :"<<vt->size()<<"\n";
+                mtbLastRead[accelerometer_runtime_index] = *vt;
+                yarp::os::Stamp info;
+                portsMTBSensors[accelerometer_runtime_index]->getEnvelope(info);
+                mtbStampLastRead[accelerometer_runtime_index] = info.getTime();
+            }
+            if( stamps != 0 )
+            {
+                *stamps = mtbStampLastRead[accelerometer_runtime_index];
+            }
+//             std::cout<<"Reached here\n";
+//             std::cout<<"------------ Read data size :"<<mtbLastRead[accelerometer_runtime_index].toString();
+//             std::cout<<"------------ Read Accelerometer Data : \n"<<vt->toString()<<", at"<<mtbStampLastRead[accelerometer_index]<<"\n";
+            int mtb_first_data_index;
+            mtb_first_data_index = accelerometersReferenceIndeces[accelerometer_index].data_index-1;
+            double *tempData;
+            tempData = mtbLastRead[accelerometer_runtime_index].data();
+            acc[0] = tempData[mtb_first_data_index];//mtbLastRead[accelerometer_imu_index][]
+            acc[1] = tempData[mtb_first_data_index+1];//TODO correct index!
+            acc[2] = tempData[mtb_first_data_index+2];//
+            ret = true;
+            break;
+//             
+        default : 
+            ret = false;
+            break;
     }
-    else
+    return ret;
+}
+
+bool yarpWholeBodySensors::readGyroscope(const int gyroscope_index, double *gyro, double *stamps, bool wait)
+{
+//     std::cout<<"----------------Trying to read gyroscope\n";
+    bool ret = true;
+    if( gyroscope_index >= (int)sensorIdList[wbi::SENSOR_GYROSCOPE].size() || gyroscope_index < 0 )
     {
-        ret = false;
+        return false;
+    }
+//     std::cout<<"----------------reference index size :"<<accelerometersReferenceIndeces.size()<<"\n";
+//     std::cout<<"----------------mtb last read size :"<<mtbLastRead.size()<<"\n";
+    int gyroscope_runtime_index;
+    switch(accelerometersReferenceIndeces[gyroscope_index].type)
```

Comment:
`accelerometersReferenceIndeces` should be `gyroscopesReferenceIndeces` ? I think that this mix happens also elsewhere in the code.